executor/cook/executor.py

```diff
@@ -0,0 +1,344 @@
+
+    Returns
+    -------
+    :return: whether the message was successfully sent
+    """
+    logging.debug('Sending framework message {}'.format(message))
+    encoded_message = encode_data(str(message).encode('utf8'))
+    if len(encoded_message) < max_message_length:
+        driver.sendFrameworkMessage(encoded_message)
+        return True
+    else:
+        log_message_template = 'Unable to send message {} as it exceeds allowed max length of {}'
+        logging.info(log_message_template.format(message, max_message_length))
+        return False
+
+
+def launch_task(driver, task, stdout_name, stderr_name):
+    """Launches the task using the command available in the data field.
+
+    Parameters
+    ----------
+    :param driver: MesosExecutorDriver
+        The mesos driver to use.
+    :param task: map
+        The task to execute.
+    :param stdout_name: string
+        The file to use to redirect stdout.
+    :param stderr_name: string
+        The file to use to redirect stderr.
+
+    Returns
+    -------
+    :return: The tuple containing the process launched, stdout file, and stderr file.
+    """
+    task_id = get_task_id(task)
+    try:
+        update_status(driver, task_id, cook.TASK_RUNNING)
+
+        command = decode_data(task['data']).decode('utf8')
+        logging.info('Command: {}'.format(command))
+        stdout = open(stdout_name, 'a+')
+        stderr = open(stderr_name, 'a+')
+        process = subprocess.Popen(command, shell=True, stdout=stdout, stderr=stderr)
+
+        return process, stdout, stderr
+    except Exception:
+        logging.exception('Error in launch_task')
+        update_status(driver, task_id, cook.TASK_FAILED)
+
+
+def is_running(process):
+    """Checks whether the process is still running.
+
+    Parameters
+    ----------
+    :param process: 
+        The process to query
+
+    Returns
+    -------
+    :return: whether the process is still running.
+    """
+    return process.poll() is None
+
+
+def cleanup_process(process_info):
+    """Closes the stdout and stderr files.
+    
+    Parameters
+    ----------
+    :param process_info: tuple
+        Tuple of (process, stdout_file, stderr_file)
+
+    Returns
+    -------
+    :return: Nothing
+    """
+    _, stdout, stderr = process_info
+    logging.info('Closing {}'.format(stdout.name))
+    stdout.close()
+    logging.info('Closing {}'.format(stderr.name))
+    stderr.close()
+
+
+def kill_task(driver, task, process_info, shutdown_grace_period_ms):
+    """Attempts to kill a process.
+     First attempt is made by sending the process a SIGTERM.
+     If the process does not terminate inside (shutdown_grace_period_ms - 100) ms, it is then sent a SIGKILL.
+     The 100 ms grace period is allocated for the executor to perform its other cleanup actions.
+
+    Parameters
+    ----------
+    :param driver: MesosExecutorDriver
+        The mesos driver to use.
+    :param task: map
+        The task to execute.
```

Comment:
nitpick: executing a task can mean both running the task and killing the task, s/execute/kill/